File: javascript/programmers/_42883.js
function solution(number, k) {
  // 시간초과... => Python Okay
  const queue = number.split("")
  stack = [queue.shift()]
  while (k && queue.length) {
    const q = queue.shift()
    while (k && stack[stack.length - 1] < q && 8) {
      stack.pop()
      k -= 1
    }
    stack.push(q)
  }
  return (k ? stack.slice(0, stack.length - k) : stack.concat(queue)).join("")
}
console.log(solution("1924", 2))
console.log(solution("1231234", 3))
console.log(solution("4177252841", 4))

File: javascript/programmers/42883.py
from collections import deque


def solution(number, k):
    queue = deque(list(number))
    stack = [queue.popleft()]
    while k and queue:
        q = queue.popleft()
        while k and stack and stack[-1] < q:
            stack.pop()
            k -= 1
        stack.append(q)
    return ''.join(stack[0:len(stack)-k]) if k else ''.join(stack + list(queue))


print(solution("1924", 2))
print(solution("1231234", 3))
print(solution("4177252841", 4))
